src/lib.rs

```diff
@@ -230,21 +228,33 @@ impl Serializable<24> for StakeInfo {
 
         let value = u64::from_reader(&mut reader)?;
         let eligibility = u64::from_reader(&mut reader)?;
-        let created_at = u64::from_reader(&mut reader)?;
+        let reward = u64::from_reader(&mut reader)?;
+        let counter = u64::from_reader(&mut reader)?;
+
+        let amount = match value > 0 {
+            true => Some((value, eligibility)),
+            false => None,
+        };
 
         Ok(Self {
-            value,
-            eligibility,
-            created_at,
+            amount,
+            reward,
+            counter,
         })
     }
 
+    /// Serializes the amount and the eligibility first, and then the reward and
+    /// the counter. If `amount` is `None`, and since a stake of no value should
+    /// not be possible, the first 16 bytes are filled with zeros.
     fn to_bytes(&self) -> [u8; Self::SIZE] {
         let mut buf = [0u8; Self::SIZE];
 
-        buf[0..8].copy_from_slice(&self.value.to_bytes());
-        buf[8..16].copy_from_slice(&self.eligibility.to_bytes());
-        buf[16..24].copy_from_slice(&self.created_at.to_bytes());
+        let (value, eligibility) = self.amount.unwrap_or_default();
+
+        buf[0..8].copy_from_slice(&value.to_bytes());
+        buf[8..16].copy_from_slice(&eligibility.to_bytes());
+        buf[16..24].copy_from_slice(&self.reward.to_bytes());
+        buf[24..32].copy_from_slice(&self.counter.to_bytes());
```

Comment:
Can't you use the `dusk-bytes`  to write into the buffer so you don't have to deal with offsets…?